projects/plugins/boost/app/lib/critical-css/Critical_CSS_State.php

```diff
@@ -389,6 +399,24 @@ public function get_provider_urls() {
 		return $this->collate_column( 'urls' );
 	}
 
+	/**
+	 * Get the provider urls with added query parameter to indicate critical css generation.
+	 *
+	 * @return array
+	 */
+	public function get_provider_urls_with_args() {
+		$providers = $this->get_provider_urls();
+		$urls      = array();
+
+		foreach ( $providers as $provider => $provider_urls ) {
+			foreach ( $provider_urls as $url ) {
+				$urls[ $provider ][] = add_query_arg( 'jb-generate-critical-css', 'true', $url );
+			}
+		}
+
+		return $urls;
+	}
```

Comment:
This is much better now - but why do we need provider urls with added query parameter? To whom are we indicating critical css generation?

Reply:
I don't remember why. Probably the plugin turns of existing critical css if the param is present.

Cc @karthikax

Reply:
In that case - why are we only doing this for cloud css and not local css generator? Alternatively - why do this here at all and not in the cloud?

Reply:
Local css adds it in the js part. I think yes, adding it on hydra would be more consistent.

Reply:
I've removed this GET parameter from this PR, and added a PR on the HYdra side to compensate: 48-gh-Automattic/boost-cloud